### PR TITLE
Add Forum settings to Channel struct

### PR DESCRIPTION
**Pull request by ikafly144**
Add some settings of forum channel as shown docs.
https://discord.com/developers/docs/change-log#add-default-layout-setting-for-forum-channels
https://discord.com/developers/docs/change-log#default-sort-order-for-forum-channels
https://discord.com/developers/docs/resources/channel#channel-object-channel-structure

### CODE REVIEW

File: structs.go [hunk 1]
```diff
@@ -270,6 +270,30 @@ const (
...
+type ForumLayout int
+
+// Block contains known ForumLayout values.
+const (
+	// No default has been set for forum channel
+	NotSet ForumLayout = 0
+	// Display posts as a list
+	ListView ForumLayout = 1
+	//Display posts as a collection of tiles
+	GalleryView ForumLayout = 2
```
Comment (FedorLap2006) [inline structs.go:294]:
```suggestion
	// No default has been set for forum channel
	FormLayoutNotSet ForumLayout = 0
	// Display posts as a list
	FormLayoutListView ForumLayout = 1
	//Display posts as a collection of tiles
	FormLayoutGalleryView ForumLayout = 2
```

File: structs.go [hunk 2]
```diff
@@ -270,6 +270,30 @@ const (
...
 
+// SortOrderType represent type of a forum channel.
+type SortOrderType int
+
+// Block contains known SortOrderType values.
+const (
+	// 	Sort forum posts by activity
+	LatestActivity SortOrderType = 0
+	// Sort forum posts by creation time (from most recent to oldest)
+	CreationDate SortOrderType = 1
```
Comment (FedorLap2006) [inline structs.go:281]:
```suggestion
	// 	Sort forum posts by activity
	ForumSortOrderLatestActivity SortOrderType = 0
	// Sort forum posts by creation time (from most recent to oldest)
	ForumSortOrderCreationDate SortOrderType = 1
```

File: structs.go [hunk 3]
```diff
@@ -270,6 +270,30 @@ const (
 	ChannelFlagRequireTag ChannelFlags = 1 << 4
 )
 
+// SortOrderType represent type of a forum channel.
+type SortOrderType int
```
Comment (FedorLap2006) [inline structs.go:274]:
```suggestion
// ForumSortOrderType represent sort order type of a forum channel
type ForumSortOrderType int
```

File: structs.go [hunk 4]
```diff
@@ -358,6 +382,18 @@ type Channel struct {
 
 	// Emoji to use as the default reaction to a forum post.
 	DefaultReactionEmoji ForumDefaultReaction `json:"default_reaction_emoji"`
+
+	// The initial RateLimitPerUser to set on newly created threads in a channel.
+	// This field is copied to the thread at creation time and does not live update.
+	DefaultRateLimitPerUser int `json:"default_rate_limit_per_user"`
```
Comment (FedorLap2006) [inline structs.go:388]: As far as I'm aware, and according to the docs, this field does not exist.
```suggestion
```

File: structs.go [hunk 5]
```diff
@@ -358,6 +382,18 @@ type Channel struct {
...
+
+	// The default sort order type used to order posts in forum channels.
+	// Defaults to null, which indicates a preferred sort order hasn't been set by a channel admin.
+	DefaultSortOrder SortOrderType `json:"default_sort_order"`
```
Comment (FedorLap2006) [inline structs.go:392]:
```suggestion
	DefaultSortOrder *SortOrderType `json:"default_sort_order"`
```

File: structs.go [hunk 6]
```diff
@@ -358,6 +382,18 @@ type Channel struct {
...
+
+	// the default forum layout view used to display posts in forum channels.
```
Comment (FedorLap2006) [inline structs.go:394]:
```suggestion
	// The default forum layout view used to display posts in forum channels.
```

**Review by FedorLap2006 — CHANGES_REQUESTED** [review]
Besides changes I've outlined, `Channel` structure is only used for retrieving data from the API.
The fields you've added also need to be added into `ChannelEdit` structure. With `omitempty`, and if necessary, pointer type.

P.S. If a field has a possible `null` value, just leave a "nullable" TODO, and keep it a pointer.

File: structs.go (hunk 4 above)
Comment (ikafly144) [reply structs.go]: Sorry, I mistyped the field name, it should have been "default_thread_rate_limit_per_user"

File: structs.go (hunk 1 above)
Comment (ikafly144) [reply structs.go]: Fixed

File: structs.go (hunk 2 above)
Comment (ikafly144) [reply structs.go]: Fixed

File: structs.go (hunk 3 above)
Comment (ikafly144) [reply structs.go]: Fixed

File: structs.go (hunk 5 above)
Comment (ikafly144) [reply structs.go]: Fixed

File: structs.go (hunk 6 above)
Comment (ikafly144) [reply structs.go]: Fixed

File: structs.go [hunk 7]
```diff
@@ -358,6 +382,18 @@ type Channel struct {
...
+	// The initial RateLimitPerUser to set on newly created threads in a channel.
+	// This field is copied to the thread at creation time and does not live update.
+	DefaultThreadRateLimitPerUser int `json:"default_thread_rate_limit_per_user"`
+
+	// The default sort order type used to order posts in forum channels.
+	// Defaults to null, which indicates a preferred sort order hasn't been set by a channel admin.
+	DefaultSortOrder *ForumSortOrderType `json:"default_sort_order"`
+
+	// The default forum layout view used to display posts in forum channels.
+	// Defaults to 0, which indicates a layout view has not been set by a channel admin.
```
Comment (FedorLap2006) [inline structs.go:395]:
```suggestion
	// Defaults to ForumLayoutNotSet, which indicates a layout view has not been set by a channel admin.
```

File: structs.go [hunk 8]
```diff
@@ -372,16 +408,17 @@ func (c *Channel) IsThread() bool {
...
+	Topic                         string                 `json:"topic,omitempty"`
+	NSFW                          *bool                  `json:"nsfw,omitempty"`
+	Position                      int                    `json:"position"`
+	Bitrate                       int                    `json:"bitrate,omitempty"`
+	UserLimit                     int                    `json:"user_limit,omitempty"`
+	PermissionOverwrites          []*PermissionOverwrite `json:"permission_overwrites,omitempty"`
+	ParentID                      string                 `json:"parent_id,omitempty"`
+	RateLimitPerUser              *int                   `json:"rate_limit_per_user,omitempty"`
+	Flags                         *ChannelFlags          `json:"flags,omitempty"`
+	DefaultThreadRateLimitPerUser int                    `json:"default_thread_rate_limit_per_user,omitempty"`
```
Comment (FedorLap2006) [inline structs.go:421]:
```suggestion
	DefaultThreadRateLimitPerUser *int                   `json:"default_thread_rate_limit_per_user,omitempty"`
```

File: structs.go [hunk 9]
```diff
@@ -394,6 +431,8 @@ type ChannelEdit struct {
 
 	AvailableTags        *[]ForumTag           `json:"available_tags,omitempty"`
 	DefaultReactionEmoji *ForumDefaultReaction `json:"default_reaction_emoji,omitempty"`
+	DefaultSortOrder     *ForumSortOrderType   `json:"default_sort_order,omitempty"`
+	DefaultForumLayout   ForumLayout           `json:"default_forum_layout,omitempty"`
```
Comment (FedorLap2006) [inline structs.go:435]:
```suggestion
	DefaultForumLayout   *ForumLayout          `json:"default_forum_layout,omitempty"`
```

**Review by FedorLap2006 — CHANGES_REQUESTED** [review]

**Comment by FedorLap2006** [comment]
Sorry for reviewing it at such a time!
But anyhow, happy new year!

**Comment by ikafly144** [comment]
In my area, the New Year has come a little earlier.
It's still early but Happy New Year!

**Comment by FedorLap2006** [comment]
Thank you for a quick response, I'll probably merge the PR on the week.

File: structs.go [hunk 10]
```diff
@@ -270,6 +270,30 @@ const (
 	ChannelFlagRequireTag ChannelFlags = 1 << 4
 )
 
+// ForumSortOrderType represent sort order type of a forum channel.
```
Comment (FedorLap2006) [inline structs.go:273]:
```suggestion
// ForumSortOrderType represents sort order of a forum channel.
```

File: structs.go [hunk 11]
```diff
@@ -270,6 +270,30 @@ const (
...
+
+// Block contains known ForumSortOrderType values.
+const (
+	// ForumSortOrderLatestActivity sorts posts by activity.
+	ForumSortOrderLatestActivity ForumSortOrderType = 0
+	// ForumSortOrderCreationDate sorts posts by creation time (from most recent to oldest).
+	ForumSortOrderCreationDate ForumSortOrderType = 1
+)
+
+// ForumLayout represent layout of a forum channel.
```
Comment (FedorLap2006) [inline structs.go:284]:
```suggestion
// ForumLayout represents layout of a forum channel.
```

File: structs.go [hunk 12]
```diff
@@ -270,6 +270,30 @@ const (
...
+type ForumLayout int
+
+// Block contains known ForumLayout values.
```
Comment (FedorLap2006) [inline structs.go:287]:
```suggestion
```

File: structs.go [hunk 13]
```diff
@@ -270,6 +270,30 @@ const (
 	ChannelFlagRequireTag ChannelFlags = 1 << 4
 )
 
+// ForumSortOrderType represent sort order type of a forum channel.
+type ForumSortOrderType int
+
+// Block contains known ForumSortOrderType values.
```
Comment (FedorLap2006) [inline structs.go:276]:
```suggestion
```

**Review by FedorLap2006 — APPROVED** [review]

**Pull request merged**